src/transformers/models/bert/modeling_bert.py

```diff
@@ -962,7 +962,7 @@ class BertModel(BertPreTrainedModel):
     `add_cross_attention` set to `True`; an `encoder_hidden_states` is then expected as an input to the forward pass.
     """
 
-    _no_split_modules = ["BertEmbeddings"]
+    _no_split_modules = ["BertEmbeddings", "BertLayer"]
```

Comment:
Is this a general solution that will work for everyone? (I mean why was this failing the test)

Reply:
Yes that will work for everyone ! I don't know why the layer were not added to `_no_split_modules`, this is what we do for all models.